File: FsUniversityApi/Database/Entities/Professor.fs
namespace FsUniversityApi.Database.Entities

//open System.ComponentModel.DataAnnotations
//open System
//open Person
//open FsUniversityApi.Database.Entities.ISoftRemovableEntity

//module Professor = 
//    [<CLIMutable>]
//        type Professor =
//            {
//                [<Key>]
//                PersonInfoId : Guid
//                PersonInfo : PersonInfo

//                [<Required>]
//                IdCode : string

//                [<Required>]
//                Subject : string

//                FirstDayAtJob : DateTime
//                Salary : int

//                Courses: Courses
//            }
//            interface IPerson with
//                member x.PersonInfo = x.PersonInfo
//            interface ISoftRemovableEntity with
//                member x.SoftDeleted = x.PersonInfo.SoftDeleted
//                member x.SetSoftDeleted flag = { x with PersonInfo = { x.PersonInfo with SoftDeleted = flag } }
                

File: GrpcService/Models/Professor.cs
namespace GrpcService.Models
{
    public class Professor
    {
        public PersonalData PersonalData { get; set; } = new PersonalData();
        public string IdCode { get; set; } = string.Empty;
        public string Subject { get; set; } = string.Empty;
        public DateTime FirstDayAtJob { get; set; } = DateTime.Now;
        public int Salary { get; set; } = 0;
    }
}
